presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -409,6 +416,8 @@ public TaskInfo updateTask(
             }
         }
 
+        queryContext.setVerboseExceededMemoryLimitErrorsEnabled(isVerboseExceededMemoryLimitErrorsEnabled(session));
+
```

Comment:
Wondering why is this done in this way rather than passing it while creating QueryContext object?

Reply:
Unfortunately there's a design flaw in how we create `QueryContext`. In Presto classic the `QueryContext` object is shared between all tasks of the same query and is created when the first task of a query is scheduled on the node. While in practice 2 tasks of the same query shouldn't have different session properties the interface allows it. Since the hack ... 

We are already using a similar hack for setting memory limits: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java#L405

Perhaps we need to refactor it at some point. But that is probably beyond the scope of this PR.